rubrics/tool/src/main/frontend/webcomponents/js/sakai-rubric.js

```diff
@@ -162,6 +200,94 @@ export class SakaiRubric extends RubricsElement {
     this.updateRubric();
   }
 
+  handleCriterionCreated(e) {
+    var data = e.detail;
+    this.rubric.criterions.push(data.criterion);
+  }
+
+  handleSaveWeights(e) {
+    var _this = this;
+    $(`[rubric-id=${this.rubric.id}]`).find('.save-weights').prop('disabled', true);
+    this.rubric.criterions.forEach(cr => {
+      cr.weight = parseFloat(cr.weight);
+      $.ajax({
+        url: `/rubrics-service/rest/criterions/${cr.id}`,
+        headers: { "authorization": this.token },
+        method: "PATCH",
+        contentType: "application/json",
+        data: JSON.stringify({ weight: cr.weight })
+      }).done(data => {
+        $(`[rubric-id=${_this.rubric.id}]`).find('.save-success').fadeIn('100');
+        setTimeout(function () {
+          $(`[rubric-id=${_this.rubric.id}]`).find('.save-weights').prop('disabled', false);
+        }, 1000);
+        setTimeout(function () {
+          $(`[rubric-id=${_this.rubric.id}]`).find('.save-success').fadeOut('100');
+        }, 5000);
+        this.requestUpdate();
+      }).fail((jqXHR, error, message) => {
+        console.log(error);console.log(message);
+      });
+    });
+  }
+
+  handleCriterionWeightChange(e){
+    var payload = e.detail;
+
+    this.rubric.criterions = payload.criteria;
+    var criterionModified = this.rubric.criterions.find(el => el.id === payload.criterionId);
+    var oldValue = criterionModified.weight;
+    var criterionId = criterionModified.id;
+
+    payload.value = payload.value.toLocaleString(this.locale);
+    criterionModified.weight = payload.value;
+    if (oldValue === payload.value) {
+      return;
+    }
+
+    var total = this.rubric.criterions.reduce((acc, el) => acc + parseFloat(el.weight.toLocaleString(this.locale)), 0);
+
+    if (total == 100) {
+      $(`[rubric-id=${this.rubric.id}]`).find('.criterion-detail .form-group').removeClass('has-error');
+      $(`[rubric-id=${this.rubric.id}]`).find('.total-weight').removeClass('has-error');
+      this.validWeight = true;
+    } else {
+      $(`[rubric-id=${this.rubric.id}]`).find('.criterion-detail .form-group').addClass('has-error');
+      $(`[rubric-id=${this.rubric.id}]`).find('.total-weight').addClass('has-error');
```

Comment:
If you can avoid modding the dom outside of lit-element, you should. Could you just add a property to the class that controls the addition of this class?